src/mscorlib/shared/System/Boolean.cs

```diff
@@ -178,23 +178,33 @@ public static Boolean Parse(String value)
             }
         }
 
+        public static bool Parse(ReadOnlySpan<char> value) =>
+            TryParse(value, out bool result) ? result : throw new FormatException(SR.Format_BadBoolean);
+
         // Determines whether a String represents true or false.
         // 
         public static Boolean TryParse(String value, out Boolean result)
         {
-            result = false;
             if (value == null)
             {
+                result = false;
                 return false;
             }
-            // For perf reasons, let's first see if they're equal, then do the
-            // trim to get rid of white space, and check again.
-            if (TrueLiteral.Equals(value, StringComparison.OrdinalIgnoreCase))
+
+            return TryParse(value.AsSpan(), out result);
+        }
+
+        public static bool TryParse(ReadOnlySpan<char> value, out bool result)
+        {
+            ReadOnlySpan<char> trueSpan = TrueLiteral.AsSpan();
```

Comment:
It is fast operation, and you cannot cache Span in a static even if you wanted to - Span cannot be assigned to static field.